ddtrace/appsec/processor.py

```diff
@@ -204,9 +207,46 @@ def __attrs_post_init__(self):
         # we always need the response headers
         self._mark_needed(_Addresses.SERVER_RESPONSE_HEADERS_NO_COOKIES)
 
-    def on_span_start(self, span):
-        # type: (Span) -> None
-        pass
+    def update_rules(self, new_rules):
+        # type: (List[Dict[str, Any]]) -> None
+        self._ddwaf.update_rules(new_rules)
+
+    def on_span_start(self, span, *args, **kwargs):
+        # type: (Span, Any, Any) -> None
+        peer_ip = kwargs.get("peer_ip")
+        headers = kwargs.get("headers", {})
+        headers_case_sensitive = bool(kwargs.get("headers_case_sensitive"))
+
+        _context.set_items(
+            {
+                "http.request.headers": headers,
+                "http.request.headers_case_sensitive": headers_case_sensitive,
+            },
+            span=span,
+        )
+
+        if peer_ip or headers:
+            ip = trace_utils._get_request_header_client_ip(span, headers, peer_ip, headers_case_sensitive)
+            # Save the IP and headers in the context so the retrieval can be skipped later
+            _context.set_item("http.request.remote_ip", ip, span=span)
+            if ip and self._is_needed(_Addresses.HTTP_CLIENT_IP):
+                data = {_Addresses.HTTP_CLIENT_IP: ip}
+                ddwaf_result = self._run_ddwaf(data)
+
+                if ddwaf_result and ddwaf_result.actions:
+                    if "block" in ddwaf_result.actions:
```

Comment:
nit: looks like these two if could be merged

Reply:
Yup. Originally it has an else for debugging but now that it doesn't have it they can be joined.